portal/src/main/webapp/js/src/url_based_content.js

```diff
@@ -0,0 +1,229 @@
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//                       VIEWS
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/**
+ * Main view
+ */
+var GeneratedPageMainView = Backbone.View.extend({
+    initialize : function (options) {
+        this.servicePresenter = options.servicePresenter;
+        this.baseURL = options.baseURL;
+        this.markdownDocumentation = options.markdownDocumentation;
+        this.hash='';
+        this.render();
+    },
+    events : {
+        "click a" : "linkClickedHandler"
+    },
+
+    linkClickedHandler: function(event){
+        // prevent link from opening
+        event.preventDefault();
+        var target = event.currentTarget.getAttribute("href");
+        this.hash = event.currentTarget.hash;
+
+        // if target starts with http, open a new tab and navigate to the target
+        if(target.startsWith("http")) window.open(target);
+        // if target starts with a hash, stay on the current page and scroll to the hash
+        else if(target.startsWith("#")){
+            this.scrollToHash();
+        }
+        else{
+            // if the target is internal, check whether the target has a hash; if so, remove it
+            if(this.hash.length>0) target = target.substring(0, target.indexOf('#'));
```

Comment:
It's my opinion that one line if/else statements are asking for maintenance trouble down the line when you accidentally forget to put in brackets later. Also the irregularity makes it slightly harder to read. But that's mostly taste.